software/bootcamp-code/main.py

```diff
@@ -0,0 +1,122 @@
+import image_processor
+import camera
+import motion
+import cv2
+import time
+import serial.tools.list_ports
+
+def main_loop():
+    debug = True
+    
+    motion_sim = motion.TurtleRobot()
+    motion_sim2 = motion.TurtleOmniRobot()
+
+    omni_motion = motion.OmniMotionRobot()
+    
+    #camera instance for normal web cameras
+    #cam = camera.OpenCVCamera(id = 2)
+    # camera instance for realsense cameras
+    cam = camera.RealsenseCamera(exposure = 100)
+    
+    processor = image_processor.ImageProcessor(cam, debug=debug)
+
+    processor.start()
+    motion_sim.open()
+    motion_sim2.open()
+
+    # incase permission issues: sudo chmod 666 /dev/tty0
+    # idProduct=5740 to filter the list given by serial.tools.list_ports.comports(include_links=False)
+    # serial connection port
+    port = '/dev/ttyACM0'
+    # get all the available seiral ports
+    ports = serial.tools.list_ports.comports()
+    for p in ports:
+        print(p.device)
+        print(p.hwid)
+        if p.hwid.__contains__("5740"):
+            port=p.device
+            print("Serial port assigned")
+            break
+
+    omni_motion.open(port)
```

Comment:
Determining the port should happen in the motion class and should not accept a port as an argument.